core/src/test/java/org/elasticsearch/action/update/UpdateResponseTests.java

```diff
@@ -0,0 +1,137 @@
+            assertEquals("{\"_index\":\"books\",\"_type\":\"book\",\"_id\":\"1\",\"_version\":2,\"result\":\"updated\"," +
+                    "\"_shards\":{\"total\":3,\"successful\":2,\"failed\":0},\"_seq_no\":7,\"get\":{\"found\":true," +
+                    "\"_source\":{\"title\":\"Book title\",\"isbn\":\"ABC-123\"},\"fields\":{\"isbn\":[\"ABC-123\"],\"title\":[\"Book " +
+                    "title\"]}}}", output);
+        }
+    }
+
+    public void testToAndFromXContent() throws IOException {
+        final XContentType xContentType = randomFrom(XContentType.values());
+        final Tuple<UpdateResponse, UpdateResponse> tuple = randomUpdateResponse();
+
+        // Parse the XContent bytes to obtain a parsed UpdateResponse
+        UpdateResponse parsedUpdateResponse;
+        try (XContentParser parser = createParser(xContentType.xContent(), toXContent(tuple.v1(), xContentType))) {
+            parsedUpdateResponse = UpdateResponse.fromXContent(parser);
+            assertNull(parser.nextToken());
+        }
+
+        final UpdateResponse updateResponse = tuple.v2();
+        try (XContentParser parser = createParser(xContentType.xContent(), toXContent(parsedUpdateResponse, xContentType))) {
+            IndexResponseTests.assertDocWriteResponse(updateResponse, parser.map());
+        }
+        assertEquals(updateResponse.getGetResult(), parsedUpdateResponse.getGetResult());
+    }
+
+    private static Tuple<UpdateResponse, UpdateResponse> randomUpdateResponse() {
+        Tuple<GetResult, GetResult> getResults = GetResultTests.randomGetResult(randomFrom(XContentType.values()));
+        GetResult actualGetResult = getResults.v1();
+
+        ShardId shardId = new ShardId(actualGetResult.getIndex(), randomAsciiOfLength(5), randomIntBetween(0, 5));
+        String type = actualGetResult.getType();
+        String id = actualGetResult.getId();
+        long version = actualGetResult.getVersion();
+        DocWriteResponse.Result result = actualGetResult.isExists() ? DocWriteResponse.Result.UPDATED : DocWriteResponse.Result.NOT_FOUND;
+
+        Long seqNo = randomBoolean() ? null : randomNonNegativeLong();
+
+        UpdateResponse actual, expected;
+        if (seqNo != null) {
+            ReplicationResponse.ShardInfo shardInfo = RandomObjects.randomShardInfo(random(), true);
+
+            actual = new UpdateResponse(shardInfo, shardId, type, id, seqNo, version, result);
+            actual.setGetResult(actualGetResult);
+
+            expected = new UpdateResponse(shardInfo, shardId, type, id, seqNo, version, result);
+
+            // We don't expect to retrieve the index/type/id of the GetResult because they are not rendered
+            // by the toXContentEmbedded method.
+            GetResult expectedGetResult = new GetResult(null, null, null, -1,
+                    getResults.v2().isExists(), getResults.v2().sourceRef(), getResults.v2().getFields());
+            expected.setGetResult(expectedGetResult);
```

Comment:
Good catch, it should be aligned with actualGetResult.isExists() / expectedGetResult.isExists(). I'll change that.